src/Symfony/Component/DependencyInjection/Compiler/ResolveInvalidReferencesPass.php

```diff
@@ -100,6 +105,11 @@ private function processArguments(array $arguments, $inMethodCall = false)
             }
         }
 
+        // Ensure numerically indexed arguments have sequential numeric keys.
+        if ($inCollection && ctype_digit(implode('', array_keys($arguments)))) {
```

Comment:
This won't replace the `ctype_digit` check, as `$isNumeric` as suggested above will return true for associative arrays.